apollo-router/src/plugins/connectors/handle_responses.rs

```diff
@@ -126,17 +128,27 @@ impl RawResponse {
                 debug_request,
                 data,
             } => {
-                let error = FetchError::SubrequestHttpError {
-                    status_code: Some(parts.status.as_u16()),
-                    service: connector.id.label.clone(),
-                    reason: format!(
-                        "{}: {}",
-                        parts.status.as_str(),
-                        parts.status.canonical_reason().unwrap_or("Unknown")
-                    ),
-                }
-                .to_graphql_error(Some((&key).into()))
-                .add_subgraph_name(&connector.id.subgraph_name);
+                let error = graphql::Error::builder()
```

Comment:
I'm not sure I understand the motivation for using `graphql::Error` here. This will just get converted right back into a `SubrequestHttperror` in `FetchService::map_to_graphql_error`, and then get converted *again* into `graphql::Error`.

Reply:
I don’t believe that’s true. These errors are added directly to the “errors” list on the response, not returned in a Result. The FetchError handling and SubrequstHttpError conversion happens in `.map_err` and related methods, which will never see these errors. 

In general, handle_response.rs code only returns Err if something really unexpected went wrong, not if a request fails.

Reply:
Ah, right, I forgot about how it wraps the error in `Ok`.